### DISCUSSION TITLE
Deleting directories on OneDrive

### DISCUSSION THREAD

**Discussion by arr19** [PowerShell/PowerShell#16110] [Q&A] [answered]
I have some directories I'm trying to delete on OneDrive using PS7.1.4
![image](https://user-images.githubusercontent.com/65091237/133543738-e69343f9-fe2e-42fd-a687-7a81c2d50aaf.png)

If I copy them to the local hard drive,
PS C:\Users\user> cd "C:\Users\user\Test - Copy"
PS C:\Users\user\Test - Copy> gci -Directory | Remove-Item -Recurse

No issues, they delete. But in OneDrive,
PS C:\Users\user\Test - Copy> cd '..\OneDrive\Test - Copy\'
PS C:\Users\user\OneDrive\Test - Copy> gci -Directory | Remove-Item -Recurse 
*Remove-Item: Cannot remove item C:\Users\user\OneDrive\Test - Copy\Local: The directory is not empty. :* 'C:\Users\user\OneDrive\Test - Copy\Local'
*Remove-Item: Cannot remove item C:\Users\user\OneDrive\Test - Copy\Online: The directory is not empty. :* 

Same message if I use the -force parameter and they are not deleted. Using PS5 gives messages about reparse-points. I've seen discussion on this saying you can use "cmd /c rmdir /s /q filename" but is there a way I don't have to rewrite code that was working before putting the files on OneDrive? There is no issue with deleting files, just directories.

**Comment by iSazonov** [comment, marked as answer]
This was fixed in PowerShell 7.2 - try latest preview build.

**Reply by arr19** [reply]
Wow, no way! That is great it got fixed right when I needed it. Thanks!
